src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -126,6 +130,370 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry(tamperedFileName);
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    using (Stream source = e.Open())
+                    {
+                        source.CopyTo(ms);
+                        Assert.True(e.Length == ms.Length);     // Only allow to decompress up to uncompressed size
+                        byte[] buffer = new byte[bufferSize];
+                        Assert.Equal(0, source.Read(buffer, 0, buffer.Length)); // shouldn't be able read more                        
+                        ms.Seek(0, SeekOrigin.Begin);
+                        int read;
+                        while ((read = ms.Read(buffer, 0, buffer.Length)) != 0)
+                        { // No need to do anything, just making sure all bytes readable
+                        }
+                        Assert.Equal(ms.Position, ms.Length); // all bytes must be read
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("normal.zip")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for full framework.")]
+        public static async Task ZipArchiveEntry_Read_CorruptedFile(string zipname)
+        {
+            string filename = zfile(zipname);
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(filename);
+            string tamperedFileName = "binary.wmv";
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry(tamperedFileName);
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    using (Stream source = e.Open())
+                    {
+                        byte[] buffer = new byte[bufferSize];
+                        int read;
+                        while ((read = source.Read(buffer, 0, buffer.Length)) != 0)
+                        {
+                            ms.Write(buffer, 0, read);
+                        }
+                        Assert.True(e.Length == ms.Length);     // Only allow to decompress up to uncompressed size
```

Comment:
All of these tests look to me like they are checking that we truncated.  Are there any tests where the deflated output is LESS than the value from the ZipArchiveEntry?

Reply:
There is no such test, but as having uncompressed size more than inlfated output is not a treat, so i think we don't need test that for reading the output stream. But that could be good case for checking local/central directory headers

Reply:
We should have a test so that we understand our behavior in that situation.  It may not need to be handled specially, but we should have a test that proves it works (or fails, whatever it does) and then that test would fail / be intentionally changed if we intentionally or accidentally change the behavior in the future.

Reply:
Added a test case